java/ql/src/semmle/code/java/frameworks/Servlets.qll

```diff
@@ -322,3 +322,21 @@ class ServletWebXMLListenerType extends RefType {
     //  - `HttpSessionBindingListener`
   }
 }
+
+/** Holds if `ma` is a method access to some override of methods of `HttpServlet`, for example `doGet` or `doPost`. */
+predicate isServletMethod(MethodAccess ma, string methodName) {
```

Comment:
```suggestion
/** Holds if `ma` is a call that gets a request parameter, called within a request handler named `methodName` (for example `doGet` or `doPost`). */
predicate getsRequestParameterInHandler(MethodAccess ma, string methodName) {
```

Reply:
I will try to improve it. Just to confirm - the scenario we want to detect is:
<code>doGet(request, response)</code> servlet method call (precondition of this query) -> <code>any_method(request)</code> -> <code>String password = request.getParameter("password")</code>.

Is my understanding correct? Please confirm. Thanks.

Reply:
Something like that, yes. For example, `doGet(request, response) { String password = getPassword(request); } ... String getPassword(request) { return request.getParameter("password"); }`

Reply:
I've modified the query to take the request variable as the source and added a new test case. Please review. Thanks.